.github/scripts/check-header.py

```diff
@@ -164,7 +169,8 @@ def main():
     checker.ignoreExtension('.vcxproj',
```

Comment:
It would make more sense to me if we only checked C files, and then manually ignore directories and files that are not to be updated with our header.

Reply:
I acknowledge this is not really within the scope of the PR, but should you want to I think that would be a good change.

Reply:
I agree with the suggestion. Tagging @dachalco to inform about the improvement suggestion.